lib/pos_consensus_controller.go

```diff
@@ -90,35 +128,16 @@ func (cc *ConsensusController) handleBlockProposerEvent(
 	event *consensus.FastHotStuffEvent,
 	expectedEventType consensus.FastHotStuffEventType,
 ) error {
-	// Validate that the expected event type is a block proposal event type
-	possibleExpectedEventTypes := []consensus.FastHotStuffEventType{
-		consensus.FastHotStuffEventTypeConstructVoteQC,
-		consensus.FastHotStuffEventTypeConstructTimeoutQC,
-	}
-	if !collections.Contains(possibleExpectedEventTypes, expectedEventType) {
-		return errors.Errorf("Invalid expected event type: %v", expectedEventType)
+	// Validate that the event's type is the expected proposal event type
+	if !isValidateBlockProposalEvent(event, expectedEventType) {
```

Comment:
```suggestion
	if !isValidBlockProposalEvent(event, expectedEventType) {
```
Nit on naming of the function